src/Metric/MetricFactory.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Baldinof\RoadRunnerBundle\Metric;
+
+use Baldinof\RoadRunnerBundle\Exception\BadConfigurationException;
+use Baldinof\RoadRunnerBundle\Exception\UnknownRpcTransportException;
+use Spiral\Goridge\RPC;
+use Spiral\Goridge\SocketRelay;
+use Spiral\RoadRunner\Metrics;
+use Spiral\RoadRunner\MetricsInterface;
+
+final class MetricFactory
+{
+    /**
+     * @var bool
+     */
+    private $metricsEnabled;
+
+    /**
+     * @var string
+     */
+    private $rrRpc;
+
+    /**
+     * @var string
+     */
+    private $kernelProjectDir;
+
+    public function __construct(string $rrRpc, string $kernelProjectDir, bool $metricsEnabled)
```

Comment:
Symfony will pass `null` if the variable does not exist. I had to disable RPC in `.rr.yml` because it's enabled by default.
```suggestion
    public function __construct(?string $rrRpc, string $kernelProjectDir, bool $metricsEnabled)
```